src/main/java/appeng/block/AEBaseBlock.java

```diff
@@ -486,36 +416,97 @@ public EnumFacing mapRotation( final IOrientable ori, final EnumFacing dir )
 		return null;
 	}
 
-	public boolean isFullSize()
+	@Override
+	public String toString()
 	{
-		return this.isFullSize;
+		String regName = getRegistryName() != null ? getRegistryName().getResourcePath() : "unregistered";
+		return getClass().getSimpleName() + "[" + regName + "]";
 	}
 
-	public boolean setFullSize( final boolean isFullSize )
+	protected String getUnlocalizedName( final ItemStack is )
 	{
-		this.isFullSize = isFullSize;
-		return isFullSize;
+		return this.getUnlocalizedName();
+	}
+
+	protected boolean hasCustomRotation()
+	{
+		return false;
+	}
+
+	protected void customRotateBlock( final IOrientable rotatable, final EnumFacing axis )
+	{
+
+	}
+
+	protected IOrientable getOrientable( final IBlockAccess w, final BlockPos pos )
+	{
+		if( this instanceof IOrientableBlock )
+		{
+			IOrientableBlock orientable = (IOrientableBlock) this;
+			return orientable.getOrientable( w, pos );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) There is an apparent infinite recursive loop in appeng.block.AEBaseBlock.getOrientable(IBlockAccess, BlockPos) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=findbugs%3AIL_INFINITE_RECURSIVE_LOOP)